src/store/authSlice.ts

```diff
@@ -0,0 +1,124 @@
+import { createSlice, createAsyncThunk, PayloadAction } from "@reduxjs/toolkit";
+import axios from 'axios';
+import { AxiosError } from 'axios';
+
+interface signUp{
```

Comment:
If this is data that is passed to the API, Consider naming the interface SignUpData